### DISCUSSION TITLE
How do game states work?

### DISCUSSION THREAD

**Discussion by eisegesis-623** [nathanhoad/godot_dialogue_manager#408] [Q&A] [answered]
Apologies for the question, but I wasn't really able to find much about the game states in the documentation. How do they work? I tried, for example, saying the following on a line of dialogue.
`do DialogueBalloon.my_custom_function()`
(My DialogueBalloon has the `temporary_game_states =  [self] + extra_game_states` and `self.dialogue_line = await resource.get_next_dialogue_line(title, temporary_game_states)` lines.)

How does this system work? Can I pass any node in the Scene Tree as a state for use in dialogue? What if method names are the same on multiple states?

(Just in case it's relevant, I'm on v2.27.3. Haven't updated yet.)

**Comment by nathanhoad** [comment, marked as answer]
Yeah the states passed in can be any node (or a dictionary) that is in scope when you pass it in. They are checked in the same order they are given so if multiple nodes have the same property or method then the first one found will be called.
